enhancements/installer/gcp-private-clusters.md

```diff
@@ -0,0 +1,197 @@
+
+Specifically for GCP, this enhancement requires implementation of internal load balancers, which were not an initial feature on the GCP platform.
+
+## Motivation
+
+### Goals
+
+Install an OpenShift cluster on GCP as internal/private so that it is only accessible from within an internal network and not visible to the public Internet.
+
+### Non-Goals
+
+There is no day-two operation for making an existing cluster private. No additional isolation from clusters in the network than what is provided by shared networking.
+
+## Proposal
+
+Installer allow users to provide a list of subnets that should be used for the cluster. Since there is an expectation that networking is being shared, the installer cannot modify the networking setup (i.e. the route tables for the subnets or the VPC options like DHCP etc.) but, changes required to the shared resources like Tags that do not affect the behavior for other tenants of the network will be made.
+
+The installer validates the assumptions about the networking setup.
+
+Infrastructure resources that are specific to the cluster and resources owned by the cluster will be created by the installer. So resources like load balancers, storage buckets, VM instances and firewall rules remain cluster managed.
+
+The infrastructure resources owned by the cluster continue to be clearly identifiable and distinct from other resources.
+
+Destroying a cluster must make sure that no resources are deleted that didn't belong to the cluster. Leaking resources is preferred over any possibility of deleting non-cluster resources.
+
+### User Stories
+
+#### Story 1
+
+As an administrator, I would like to install an OpenShift cluster in my company's private network that is accessible to within the network but not on the Internet.
+
+### Implementation Details
+
+#### Resources Provided to the Installer
+
+Users provide a VPC and subnets in the install config `platform` and specify an `internal` publishing strategy in the top-level `publish` attribute:
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: example
+platform:
+  gcp:
+    projectID: example-project
+    region: us-east4
+    computeSubnet: example-worker-subnet
+    controlPlaneSubnet: example-master-subnet
+    network: example-network
+publish: Internal
+pullSecret: '{"auths": ...}'
+```
+
+##### Basedomain
+
+`Basedomain` will continue to be used for creating a private DNS zone and the necessary records, but the requirement for a public DNS zone for the basedomain will be removed.
+
+##### Public Subnets
+
+No public subnets need to be provided to the installer.
```

Comment:
on GCP there is no public vs private subnet, only control-plane and compute ??

Reply:
I removed this section.